### DISCUSSION TITLE
Add extra database column(field) after user create a table(data model)

### DISCUSSION THREAD

**Discussion by phxgbl** [directus/directus#6917] [Feature Requests]
I have a requirement where when a user creates a data model, an **extra column** 'tenant_id' should be added, for achieving single DB multi-tenancy. I tried creating a role and assigning **field preset** to the collection like { "tenant_id": "$CURRENT_USER" } . So what I expect is when I create a new collection object, a column tenant_id should be added with the value of logged-in user which doesn't happen. Any insights on that?

![Screenshot from 2021-07-21 22-29-10](https://user-images.githubusercontent.com/37118923/126580636-cc1feaca-38d9-4b59-88d7-d5dd8a546b86.png)
![Screenshot from 2021-07-21 22-31-09](https://user-images.githubusercontent.com/37118923/126580637-b7393106-9f05-4753-b53b-cb6ee5c1d6a8.png)
![Screenshot from 2021-07-21 22-31-57](https://user-images.githubusercontent.com/37118923/126580640-07dc91b7-f7d1-40cd-bb29-4528d49272a5.png)

**Comment by rijkvanzanten** [comment]
Are you trying to add this `tenant_id` column to `directus_collections` itself? I'm not 100% clear on what you're trying to achieve exactly 🤔

**Reply by phxgbl** [reply]
@rijkvanzanten I want to accomplish multi tenancy, so every user will be associated with a tenant and will have a tenant id, so if he creates a data model, tenant_id column should be added by default. Is it possible?

**Reply by rijkvanzanten** [reply]
> so if he creates a data model

That's the part I'm a little confused about. Are you saying when the user adds a new item to an existing collection, or when the user _creates_ a new collection?

**Reply by phxgbl** [reply]
@rijkvanzanten new collection..

**Reply by rijkvanzanten** [reply]
Ah ok! I don't think it'll work as you expect 🤔 You have to be an admin user in order to modify the schema, and admin users have full permissions to everything at all times. Therefore, you can't have a user create their own collections without seeing the other collections as well.

**Reply by phxgbl** [reply]
@rijkvanzanten got it.. Thanks a lot
